hackathon_site/dashboard/frontend/src/components/general/Navbar/Navbar.js

```diff
@@ -11,73 +12,81 @@ import { ReactComponent as Inventory } from "assets/images/icons/Hardware.svg";
 // Components
 import Button from "@material-ui/core/Button";
 
-const Navbar = ({ cartQuantity }) => {
-    return (
-        <nav className={styles.nav}>
-            <div className={styles.navFlexDiv}>
-                <Link to={"/"}>
-                    <Button
-                        className={styles.navBtn}
-                        aria-label="Dashboard"
-                        startIcon={<DashboardIcon />}
-                    >
-                        Dashboard
-                    </Button>
-                </Link>
-                <Link to={"/orders"}>
-                    <Button
-                        className={styles.navBtn}
-                        aria-label="Orders"
-                        startIcon={<ListAlt />}
-                    >
-                        Orders
-                    </Button>
-                </Link>
-                <Link to={"/teams"}>
-                    <Button
-                        className={styles.navBtn}
-                        aria-label="Teams"
-                        startIcon={<AccountBoxIcon />}
-                    >
-                        Teams
-                    </Button>
-                </Link>
-                <Link to={"/reports"}>
-                    <Button
-                        className={styles.navBtn}
-                        aria-label="Reports"
-                        startIcon={<InsertChartOutlined />}
-                    >
-                        Reports
-                    </Button>
-                </Link>
-                <Link to={"/inventory"}>
-                    <Button
-                        className={styles.navBtn}
-                        aria-label="Inventory"
-                        startIcon={<Inventory className={styles.customIcon} />}
-                    >
-                        Inventory
-                    </Button>
-                </Link>
-                <Link to={"/cart"}>
-                    <Button
-                        className={styles.navBtn}
-                        aria-label="Cart"
-                        startIcon={<LocalMallIcon />}
-                    >
-                        Cart ({cartQuantity})
-                    </Button>
-                </Link>
-            </div>
-            <Button
-                aria-label="Logout"
-                className={`${styles.navBtn} ${styles.navBtnLogout}`}
-            >
-                <b>Logout</b>
-            </Button>
-        </nav>
-    );
-};
+const UnconnectedNavbar = ({ cartQuantity, pathname }) => (
+    <nav className={styles.nav}>
+        <div className={styles.navFlexDiv}>
+            <Link to={"/"}>
+                <Button
+                    className={pathname === "/" ? styles.navActive : styles.navBtn}
```

Comment:
It would be nice to test that this is actually happening. A possibility:
```javascript
it("Adds the active class to only the active page", () => {
        const pagesAndPaths = [
            ["Dashboard", "/"],
            ["Orders", "/orders"],
            // ...
        ];

        pagesAndPaths.map(([label, path]) => {
            const { getByText, container } = render(
                withStore(withRouter(<Navbar pathname={path} />))
            );

            expect(getByText(label).closest("button").className).toMatch(
                new RegExp(styles.navActive)
            );

            expect(container.querySelectorAll(".navActive").length).toBe(1);
        });
    });
```

RTL really doesn't want you to be querying and using classes or ids in your test, since it abstracts away from what the user sees. However, in this case I think it is appropriate, and there isn't any other thing we could do.

Reply:
Had to do a
```javascript
import { styles } from "./Navbar.module.scss";
```
at the top, of course

Reply:
Seems that multiple calls to `render` without cleaning up will keep adding to the DOM, so this will fail for anything beyond the first map item. I can't get [cleanup](https://testing-library.com/docs/react-testing-library/api#cleanup) to work on my end, but you can give it a try. The other solution is to just map `pagesAndPaths` and have the `it` call with the `render` and `expect`s inside it.